### PR TITLE
leverage generateTemplates capability of harvester/addons

### PR DESCRIPTION
**Pull request by ibrokethecloud**
**Problem:**

PR introduces minor changes to installer to leverage the new `generateTemplate` capability added to `harvester/addons`

Prior to this change the addon updates were being managed by updates to version_info file, and also updating the version details in the rancher-22-addons.yaml template file.

**Solution:**

This has been simplified to use the chart versions in version_info file and allow ability to generate template file dynamically. This ensures that chart and image version change is only ever kept in one single location.

addons PR: https://github.com/harvester/addons/pull/4

**Related Issue:**
https://github.com/harvester/harvester/issues/4937
**Test plan:**

### CODE REVIEW

**Comment by ibrokethecloud** [comment]
@w13915984028 now that `version_info` is the sole driver of template and addon generation, do you think we still need the chart / addon version check in `build-bundle`?

**Review by tserong — APPROVED** [review]

File: scripts/build-bundle [hunk 1]
```diff
@@ -140,9 +145,9 @@ check_addon_chart_version_matching() {
   ls -alht ${CHARTS_DIR}
 
   echo "addon template files"
-  ls -alht ${addons_path}/pkg/templates
+  ls -alht ${addons_path}
```
Comment (Yu-Jack) [inline scripts/build-bundle:148]: Should we change this to `ls -alht ${addons_path}/*.yaml` ? I think we just care the *.yaml, right?

**Review by Vicente-Cheng — APPROVED** [review]
lgtm, thanks for the improvement

File: scripts/build-bundle (hunk 1 above)
Comment (ibrokethecloud) [reply scripts/build-bundle:148]: this is to only to help with debugging.. the subsequent loop limits itself to `*.yaml`

**Review by Yu-Jack — APPROVED** [review]

**Comment by ibrokethecloud** [comment]
@mergifyio backport v1.4

**Pull request merged**